CodingChallenges/CC_051_astar/Processing/CC_051_astar/Spot.pde

```diff
@@ -0,0 +1,88 @@
+// Daniel Shiffman
+// http://codingtra.in
+// http://patreon.com/codingtrain
+// Part 1: https://youtu.be/aKYlikFAV4k
+// Part 2: https://youtu.be/EaZxUCWAjb0
+// Part 3: https://youtu.be/jwRT4PCT6RU
+// Processing transcription: Chuck England
+
+import java.util.*;
+
+// An object to describe a spot in the grid
+class Spot {
+  int i;
+  int j;
+  // f, g, and h values for A*
+  float f = 0;
+  float g = 0;
+  float heuristic = 0;
+  
+  // Neighbors
+  List<Spot> neighbors = new ArrayList<Spot>();
+
+  // Where did I come from?
+  Spot previous = null;
+
+  boolean wall = false;
+  
+  Spot(int i_, int j_) {
+
+    // Location
+    i = i_;
+    j = j_;
+
+    // Am I a wall?
+    wall = false;
+    if (random(1) < 0.4) {
+      wall = true;
+    }
+  }
+  
+  // Display me
+  void show() {
+    if (wall) {
+      fill(0);
+      noStroke();
+      ellipse((float)i * w + w / 2.0, (float)j * h + h / 2.0, w / 2.0, h / 2.0);
```

Comment:
Is the casting here necessary?

Reply:
Hmm, I was actually hunting a bug when I did it. I think not, but I'll double check.
The issue was that the h variable in the class was shadowing the global one. So I changed the name of the variable. Removing the casts.